crates/rpc/src/error.rs

```diff
@@ -2,83 +2,215 @@ use beerus_core::CoreError;
+use starknet::providers::ProviderError;
+use starknet::providers::ProviderError::StarknetError as StarknetProviderError;
 
 #[derive(Debug)]
 pub enum BeerusRpcError {
-    Provider(ProviderError<JsonRpcClientError<HttpTransportError>>),
+    Provider(ProviderError),
     Other((i32, String)),
 }
 
 impl From<BeerusRpcError> for ErrorObjectOwned {
     fn from(err: BeerusRpcError) -> Self {
         match err {
-            BeerusRpcError::Provider(e) => {
-                match e {
-                    ProviderError::StarknetError(e) => {
-                        let (code, message) = (e.code, e.message);
-                        let code = match code {
-                        MaybeUnknownErrorCode::Known(known) => match known {
-                            StarknetError::FailedToReceiveTransaction => 1,
-                            StarknetError::ContractNotFound => 20,
-                            StarknetError::BlockNotFound => 24,
-                            StarknetError::InvalidTransactionIndex => 27,
-                            StarknetError::ClassHashNotFound => 28,
-                            StarknetError::TransactionHashNotFound => 29,
-                            StarknetError::PageSizeTooBig => 31,
-                            StarknetError::NoBlocks => 32,
-                            StarknetError::InvalidContinuationToken => 33,
-                            StarknetError::TooManyKeysInFilter => 34,
-                            StarknetError::ContractError => 40,
-                            StarknetError::ClassAlreadyDeclared => 51,
-                            StarknetError::InvalidTransactionNonce => 52,
-                            StarknetError::InsufficientMaxFee => 53,
-                            StarknetError::InsufficientAccountBalance => 54,
-                            StarknetError::ValidationFailure => 55,
-                            StarknetError::CompilationFailed => 56,
-                            StarknetError::ContractClassSizeIsTooLarge => 57,
-                            StarknetError::NonAccount => 58,
-                            StarknetError::DuplicateTx => 59,
-                            StarknetError::CompiledClassHashMismatch => 60,
-                            StarknetError::UnsupportedTxVersion => 61,
-                            StarknetError::UnsupportedContractClassVersion => 62,
-                            StarknetError::UnexpectedError => 63,
-                        },
-                        MaybeUnknownErrorCode::Unknown(unknown) => unknown as i32,
-                    };
-                        ErrorObjectOwned::owned(code, message, None::<()>)
-                    }
-                    _ => ErrorObjectOwned::owned(
-                        -32601,
-                        format!("{e}"),
+            BeerusRpcError::Provider(provider_err) => match provider_err {
+                // TODO Do we actually need to own these?
```

Comment:
Is any action necessary here before merging?

Reply:
Not really. I see allocations that could have been avoided here, but this isn't trivial nor needed.
Won't fix.